### ISSUE TITLE
Build failed: `lottie_single_player.cpp` problem(win32)

### ISSUE THREAD

**Issue by 0xNima** [telegramdesktop/tdesktop#17107]
Hi. I got this error while building the solution.
`lottie_single_player.cpp(105,37): error C2326: 'void Lottie::SinglePlayer::SharedProvider::<lambda_1>::()::<lambda_1>::operator ()(QByteArray &&) const': function cannot access 'state'`
Can anyone help me, please?

**Comment by den777-d** [comment]
I also have such a problem

**Comment by 0xNima** [comment]
> I also have such a problem (win64)
I've fixed it with a hacky workaround in [L#105](https://github.com/desktop-app/lib_lottie/blob/ad7fce76f3b403471a296c928bae67cd36b8b2cf/lottie/lottie_single_player.cpp#L105):

```cpp
const auto _state = state;
crl::async([=, done = std::move(_state->done)]() mutable {
```
but I'm not experienced in cpp and I don't know what are the side effects of that
